doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,299 @@
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |R R R R R R|              Timestamp                            |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                          Packet ID                            |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |   Padding     |    Padding    |  OptType = 4  |  OptLen = ?   |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |    ....       | | |               ....                        |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                       Path Validator                          |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+Header fields computation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. math::
+    \begin{align*}
+        &encryptedPolicyID = policyID \oplus AES.Encrypt(K_i, Identifier)[0]\\\\
+        &policyID = encryptedPolicyID \oplus AES.Encrypt(K_i, Identifier)[0]\\\\
+        &K_i = DRKey\,(AS\,A_i \rightarrow AS_0:Endhost)\,or\,(AS\,A_i \rightarrow AS_0)\\\\
+        &HVF_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[0:3] \oplus 0x3FFFFF\\\\
+        &HVFVerified_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[3:6] \oplus 0x3FFFFF\\\\
+    \end{align*}
+
+Data plane
+----------
+
+Processing at the router
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Whenever a FABRID enabled router receives a SCION packet, it has to figure out whether it should be processed as FABRID or not.
+In both cases, all the logic of a normal SCION packet will be applied too.
+The router determines whether the SCION packet is a FABRID packet as follow:
+
+.. image:: fig/FABRID/FABRIDActivation.png
+    :scale: 70%
+
+If the SCION packet uses FABRID, the router is going to verify the correctness of the current FABRID Hop-validation-field using
+either the AS-to-AS or AS-to-Host DRKey and verifies whether the encrypted policy index matches a valid FABRID policy.
+If this is the case, the router will update the FABRID Hop-validation-field accordingly and route the packet over
+an intra-AS path matching the provided FABRID policy.
+The corresponding intra-AS paths are provided to the border routers by the local control service.
+
+Processing at the endhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To be able to send a FABRID packet, the endhost has to choose a path that supports its path constraints.
```

Comment:
@jeltevanbommel I think you can better answer this question.

Reply:
Would be great to have a concrete example, e.g., a code snippet

Reply:
Ok I see, thanks for the clarification.
I think it would help to (i) explain that this is just an array for snet, and to (ii) have an example of how the parameter on top of endhost applications look like.

Reply:
```suggestion
To be able to send a FABRID packet, the endhost has to choose a path that supports its path and policy constraints. A detailed explanation on how endhost applications can find such paths is given in the section :ref:`Exposing policies to the end hosts <endhost_policy_selection>`. Once a path has been found, with specific policies for each hop in the path, the path and an array containing one policy per hop is given to the FABRID snet implementation. The snet implementation then constructs the FABRID packet by automatically requesting the necessary DRKeys and computing the hop validation fields. The packet can then be sent to the border router for further forwarding. A receiving endhost can recompute the path validator to verify that the packet was forwarded over this path.
```